test/e2e/storage/testsuites/readwriteoncepod.go

```diff
@@ -141,20 +158,33 @@ func (t *readWriteOncePodTestSuite) DefineTests(driver storageframework.TestDriv
 		framework.ExpectNoError(err, "failed to wait for pod1 running status")
 		l.pods = append(l.pods, pod1)
 
-		// Create the second pod, which will fail scheduling because the ReadWriteOncePod PVC is already in use.
+		// Create the second pod, which will preempt the first pod because it's using the
+		// ReadWriteOncePod PVC and has higher priority.
 		pod2, err := e2epod.MakeSecPod(&podConfig)
 		framework.ExpectNoError(err, "failed to create spec for pod2")
+		pod2.Spec.PriorityClassName = l.priorityClass.Name
 		_, err = l.cs.CoreV1().Pods(pod2.Namespace).Create(context.TODO(), pod2, metav1.CreateOptions{})
 		framework.ExpectNoError(err, "failed to create pod2")
-		err = e2epod.WaitForPodNameUnschedulableInNamespace(l.cs, pod2.Name, pod2.Namespace)
-		framework.ExpectNoError(err, "failed to wait for pod2 unschedulable status")
 		l.pods = append(l.pods, pod2)
 
-		// Delete the first pod and observe the second pod can now start.
-		err = e2epod.DeletePodWithWait(l.cs, pod1)
-		framework.ExpectNoError(err, "failed to delete pod1")
+		// Wait for the first pod to be preempted and the second pod to start.
+		err = e2epod.WaitForPodToDisappear(l.cs, pod1.Namespace, pod1.Name, labels.Everything(), framework.Poll, f.Timeouts.PodStart)
```

Comment:
Per this comment https://github.com/kubernetes/kubernetes/pull/114051/#issuecomment-1325639784, does this mean I'll need to add `nodeSelectors` to pod1 and pod2 to force preemption when this test runs in a multi-node environment?

Reply:
Not necessarily. If the pod2 fits in the space that pod1 would leave, then preemption should work.